UnitTests/GitUITests/UserControls/ToolStripPushButtonTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using FluentAssertions;
+using GitCommands;
+using GitCommands.Git;
+using GitUI.CommandsDialogs;
+using GitUI.Properties;
+using JetBrains.Annotations;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace GitUITests.UserControls
+{
+    [TestFixture]
+    public class ToolStripPushButtonTests
+    {
+        private static readonly int PushImageHashCode = Images.Push.RawFormat.GetHashCode();
+        private bool _originalShowAheadBehindData;
+        private IAheadBehindDataProvider _aheadBehindDataProvider;
+        private ToolStripPushButton _sut;
+
+        [OneTimeSetUp]
+        public void SetUpFixture()
+        {
+            _originalShowAheadBehindData = AppSettings.ShowAheadBehindData;
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            AppSettings.ShowAheadBehindData = _originalShowAheadBehindData;
```

Comment:
Just realised, there are no tests that set `AppSettings.ShowAheadBehindData` false